go/codegen/experimental/codegen_step.go

```diff
@@ -159,19 +161,27 @@ func isXGBoostEstimator(estimator string) bool {
 type Metadata simplejson.Json
 
 func (m *Metadata) imageName() string {
+	if m == nil { // maybe PAI ML models
+		return ""
+	}
 	return (*simplejson.Json)(m).Get("model_repo_image").MustString()
 }
 
-func getModelMetadata(session *pb.Session, table string) (*Metadata, error) {
+func getModelMetadata(session *pb.Session, modelName string) (*Metadata, error) {
 	submitter := getSubmitter(session)
-	if submitter == "local" {
-		modelZooAddr, table, tag := decomposeModelName(table)
-		if modelZooAddr != "" {
-			return getModelMetadataFromModelZoo(modelZooAddr, table, tag)
-		}
-		return GetModelMetadataFromDB(session.DbConnStr, table)
+	modelZooAddr, modelName, tag := decomposeModelName(modelName)
+	if modelZooAddr != "" {
+		return getModelMetadataFromModelZoo(modelZooAddr, modelName, tag)
+	}
+
+	switch submitter {
+	case "local":
+		return GetModelMetadataFromDB(session.DbConnStr, modelName)
+	case "pai", "pai_local", "alisa":
+		return getModelMetadataFromOSS(session, modelName)
```

Comment:
We need to save pai models into the database using the same way: 

https://github.com/sql-machine-learning/sqlflow/blob/dbefee47f00e032078e3a60ae9d803aae54b417e/go/executor/pai.go#L222-L229